File: backend/app/schemas/analytics.py
"""
运营监控、ECharts 走势与安全审计数据契约 (Analytics Schemas)

职责:
    - 大盘核心 KPI 汇总卡片响应模型 (PV/UV, 切片总量, 已发布FAQ, 待处理缺口, 响应耗时, 缓存命中率)
    - ECharts 趋势图数据序列模型 (近7天 Token 双轴、延时趋势)
    - 高频提问 TOP 5 与热门知识 TOP 5 榜单响应模型
    - 问答全链路审计日志查询与受限明细证据链下钻模型

架构定位:
    数据契约层 (Schemas) / 模块六: 运营监控与审计大盘

作者:
    System Architect (系统架构组) & Backend Team
"""

from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, ConfigDict, Field, model_validator


class DashboardSummaryResponse(BaseModel):
    """大盘 5 大核心 KPI 统计汇总响应"""
    model_config = ConfigDict(from_attributes=True)

    pv: int = Field(default=0, description="页面/提问总访问量 (Page Views)")
    uv: int = Field(default=0, description="独立访问用户数 (Unique Visitors)")
    chunks_count: int = Field(default=0, description="知识切片总量")
    total_chunks: Optional[int] = Field(default=None, description="知识切片总量 (别名)")
    knowledge_units_count: int = Field(default=0, description="知识文档/单元总量")
    published_faqs: int = Field(default=0, description="已采纳发布的标准 FAQ 总数")
    faqs_count: Optional[int] = Field(default=None, description="已发布 FAQ 总数 (别名)")
    open_gaps: int = Field(default=0, description="待处理知识缺口盲区总数")
    pending_gaps: Optional[int] = Field(default=None, description="待处理缺口 (别名)")
    knowledge_gaps_count: Optional[int] = Field(default=None, description="知识缺口总数 (别名)")
    faq_cache_hit_rate: float = Field(default=0.0, description="FAQ 极速缓存直出命中率百分比 (0.0~100.0)")
    avg_latency_ms: float = Field(default=0.0, description="端到端平均响应耗时 (ms)")
    total_tokens: int = Field(default=0, description="累计消耗 Token 总数")
    pv_uv_delta: Optional[str] = Field(default="0% 较昨日", description="PV/UV环比增量标签")
    units_synced: Optional[int] = Field(default=0, description="已同步知识单元数")
    units_pending: Optional[int] = Field(default=0, description="待更新知识单元数")
    tokens_saved: Optional[str] = Field(default="节约 0 Token", description="节约Token数描述")
    p95_latency_ms: Optional[float] = Field(default=0.0, description="P95响应延时")
    p99_latency: Optional[str] = Field(default="0ms", description="P99响应延时")
    unresolved_gaps_count: Optional[int] = Field(default=0, description="待闭环知识缺口")
    unresolved_gaps_delta: Optional[str] = Field(default="0个 本周新增", description="待闭环缺口增量")

    @model_validator(mode="after")
    def sync_aliases(self) -> "DashboardSummaryResponse":
        if self.total_chunks is None:
            self.total_chunks = self.chunks_count
        elif self.chunks_count is None:
            self.chunks_count = self.total_chunks

        if self.faqs_count is None:
            self.faqs_count = self.published_faqs
        elif self.published_faqs is None:
            self.published_faqs = self.faqs_count

        if self.pending_gaps is None:
            self.pending_gaps = self.open_gaps
        if self.knowledge_gaps_count is None:
            self.knowledge_gaps_count = self.open_gaps
        if self.unresolved_gaps_count is None or self.unresolved_gaps_count == 0:
            self.unresolved_gaps_count = self.open_gaps
        return self


# 别名兼容
DashboardSummary = DashboardSummaryResponse


class LatencyBucketResponse(BaseModel):
    """端到端响应耗时区间统计模型 (适配 ECharts/前端柱状图)"""
    label: str = Field(..., description="延时区间标签，如 < 50ms (FAQ直出)")
    percent: float = Field(default=0.0, description="占比百分比 (0.0~100.0)")
    count: int = Field(default=0, description="该区间命中日志条数")
    sub_label: Optional[str] = Field(default=None, description="子标签说明")


class TrendPoint(BaseModel):
    """每日 Token 消耗与延时趋势走势数据点"""
    date: str = Field(..., description="日期 (YYYY-MM-DD)")
    time: Optional[str] = Field(default=None, description="时间标签 (如 MM-DD 或 HH:MM)")
    prompt_tokens: int = Field(default=0, description="输入 Prompt Tokens")
    completion_tokens: int = Field(default=0, description="输出 Completion Tokens")
    total_tokens: int = Field(default=0, description="当日消耗 Token 总量")
    latency_ms: float = Field(default=0.0, description="当日平均延时毫秒")
    avg_latency_ms: Optional[float] = Field(default=None, description="当日平均延时 (别名)")
    pv: int = Field(default=0, description="当日提问量")
    qps: Optional[float] = Field(default=0.0, description="当日峰值 QPS (次/秒)")

    @model_validator(mode="after")
    def sync_latency(self) -> "TrendPoint":
        if self.avg_latency_ms is None:
            self.avg_latency_ms = self.latency_ms
        elif self.latency_ms == 0.0 and self.avg_latency_ms is not None:
            self.latency_ms = self.avg_latency_ms
        if not self.time and self.date:
            self.time = self.date[5:] if len(self.date) >= 10 else self.date
        return self


# 别名兼容
MetricTrendPoint = TrendPoint


class TopRankingItem(BaseModel):
    """排行榜数据项 (高频提问 TOP / 热门知识引用 TOP)"""
    title: str = Field(..., description="榜单项目名称/提问文本/文档标题")
    query: Optional[str] = Field(default=None, description="提问文本 (别名)")
    count: int = Field(..., description="频次/引用热度/命中次数")
    hit_count: Optional[int] = Field(default=None, description="频次 (别名)")
    rank: Optional[int] = Field(default=1, description="排名次序 (1~5)")

    @model_validator(mode="after")
    def sync_aliases(self) -> "TopRankingItem":
        if not self.query and self.title:
            self.query = self.title
        if self.hit_count is None:
            self.hit_count = self.count
        return self


class TopRankingsResponse(BaseModel):
    """双维度 TOP 5 榜单聚合响应"""
    top_queries: List[TopRankingItem] = Field(default_factory=list, description="高频提问 TOP 5")
    top_knowledge: List[TopRankingItem] = Field(default_factory=list, description="热门知识/切片引用 TOP 5")
    items: Optional[List[TopRankingItem]] = Field(default=None, description="通用项列表")

    @model_validator(mode="after")
    def sync_items(self) -> "TopRankingsResponse":
        if self.items is None:
            self.items = list(self.top_queries)
        return self


class AuditLogResponse(BaseModel):
    """全链路安全审计日志详情响应"""
    model_config = ConfigDict(from_attributes=True)

    id: int = Field(..., description="审计记录主键ID")
    trace_id: str = Field(..., description="全链路唯一追踪号 X-Trace-Id")
    conversation_id: Optional[str] = Field(default=None, description="会话ID")
    user_id: Optional[int] = Field(default=None, description="提问员工ID")
    username: Optional[str] = Field(default=None, description="员工登录账号/姓名")
    employee_id: Optional[str] = Field(default=None, description="员工工号")
    user_dept: Optional[str] = Field(default=None, description="所属部门")
    user_role: Optional[str] = Field(default=None, description="所属角色")
    dept_name: Optional[str] = Field(default=None, description="所属部门 (别名)")
    role_name: Optional[str] = Field(default=None, description="所属角色 (别名)")
    query_text: str = Field(..., description="用户提问内容")
    query: Optional[str] = Field(default=None, description="用户提问内容 (别名)")
    answer_snippet: Optional[str] = Field(default=None, description="答复摘要或兜底提示")
    recalled_chunk_ids: List[int] = Field(default_factory=list, description="向量检索初筛召回切片ID清单")
    recalled_count: int = Field(default=0, description="初筛召回切片总数")
    allowed_chunk_ids: List[int] = Field(default_factory=list, description="4D安全放行切片ID清单")
    allowed_count: int = Field(default=0, description="安全放行切片数")
    restricted_chunk_ids: List[int] = Field(default_factory=list, description="4D安全隔离/越权拦截受限切片ID清单")
    restricted_count: int = Field(default=0, description="受限切片数")
    is_blocked: bool = Field(default=False, description="是否触发拦截/受限隔离/静默回退")
    block_reason: Optional[str] = Field(default=None, description="拦截/受限成因")
    prompt_tokens: int = Field(default=0, description="输入消耗 Tokens")
    completion_tokens: int = Field(default=0, description="输出消耗 Tokens")
    total_tokens: int = Field(default=0, description="总消耗 Tokens")
    tokens: Optional[int] = Field(default=None, description="总消耗 Tokens (别名)")
    latency_ms: float = Field(default=0.0, description="调用处理耗时 (ms)")
    created_at: Optional[datetime] = Field(default=None, description="记录发生时间")
    time: Optional[str] = Field(default=None, description="格式化时间字符串")
    verdict_type: Optional[str] = Field(default=None, description="安全裁决类型 (BLOCKED/ALLOWED/FAQ_HIT)")
    verdict_badge_text: Optional[str] = Field(default=None, description="安全裁决徽章文案")
    sha256_hash: Optional[str] = Field(default=None, description="存证哈希指纹")
    silent_intercept_triggered: Optional[bool] = Field(default=False, description="是否触发静默防越权脱敏")
    evidence_chain: Optional[Dict[str, Any]] = Field(default=None, description="完整下钻安全证据链明细")

    @model_validator(mode="after")
    def sync_query(self) -> "AuditLogResponse":
        if not self.query and self.query_text:
            self.query = self.query_text
        elif not self.query_text and self.query:
            self.query_text = self.query

        if not self.dept_name and self.user_dept:
            self.dept_name = self.user_dept
        if not self.role_name and self.user_role:
            self.role_name = self.user_role

        if self.tokens is None:
            self.tokens = self.total_tokens

        if not self.time and self.created_at:
            self.time = self.created_at.strftime("%H:%M:%S")

        if not self.verdict_type:
            if self.is_blocked:
                self.verdict_type = "BLOCKED"
                self.verdict_badge_text = f"阻断 {self.restricted_count} (4D越权)"
                self.silent_intercept_triggered = True
            elif self.latency_ms < 50 and not self.recalled_chunk_ids:
                self.verdict_type = "FAQ_HIT"
                self.verdict_badge_text = "⚡ FAQ 命中直出"
            else:
                self.verdict_type = "ALLOWED"
                self.verdict_badge_text = "✓ 4D全部放行"

        if not self.sha256_hash and self.trace_id:
            import hashlib
            self.sha256_hash = hashlib.sha256(f"{self.trace_id}_{self.id}".encode()).hexdigest()
        return self


# 别名兼容
AuditLogItem = AuditLogResponse


class EvidenceChainDetailResponse(BaseModel):
    """拦截证据链下钻详情响应"""
    trace_id: str = Field(..., description="追踪号")
    query: str = Field(..., description="提问内容")
    user_context: Dict[str, Any] = Field(default_factory=dict, description="提问用户4D属性上下文")
    recalled_chunks: List[Dict[str, Any]] = Field(default_factory=list, description="初筛召回切片明细")
    allowed_chunks: List[Dict[str, Any]] = Field(default_factory=list, description="放行切片明细")
    restricted_chunks: List[Dict[str, Any]] = Field(default_factory=list, description="拦截受限切片证据链")
    decision: Dict[str, Any] = Field(default_factory=dict, description="安全裁决裁定结论")
    metrics: Dict[str, Any] = Field(default_factory=dict, description="算力与延时耗时分析")

File: front/src/api/analytics.ts
/**
 * 运营监控大盘与全链路审计流水 API
 * 模块: FE-M6 / M6: Analytics
 * 原型对应: PAGE-07 (operations_analytics.pen)
 */

import request, { generateTraceId } from '@/utils/request'
import type { ApiResponse, PaginatedData, PaginationParams } from '@/types/common'
import type {
  DashboardSummary,
  TokenTrendPoint,
  LatencyBucket,
  TopRankingsResponse,
  AuditLogItem
} from '@/types/analytics'

const isMockEnabled = (): boolean => {
  return import.meta.env.VITE_ENABLE_MOCK === 'true'
}

// 预置原型 1:1 运营指标 Mock 数据
const liveMockSummary: DashboardSummary = {
  pv: 34280,
  uv: 2840,
  pv_uv_delta: '↑ 12.4% 较昨日',
  knowledge_units_count: 128,
  chunks_count: 1420,
  units_synced: 124,
  units_pending: 4,
  faq_cache_hit_rate: '41.8%',
  tokens_saved: '节约 8.4M Token',
  avg_latency_ms: 380,
  p95_latency_ms: 820,
  p99_latency: '1.2s',
  unresolved_gaps_count: 14,
  unresolved_gaps_delta: '↑ 3个 本周新增',
  published_faqs_count: 86
}

const liveMockTokenTrends: TokenTrendPoint[] = [
  { time: '00:00', prompt_tokens: 120000, completion_tokens: 45000, qps: 8 },
  { time: '06:00', prompt_tokens: 80000, completion_tokens: 30000, qps: 4 },
  { time: '09:00', prompt_tokens: 380000, completion_tokens: 140000, qps: 32 },
  { time: '12:00', prompt_tokens: 290000, completion_tokens: 110000, qps: 24 },
  { time: '15:00', prompt_tokens: 450000, completion_tokens: 180000, qps: 45 },
  { time: '18:00', prompt_tokens: 320000, completion_tokens: 130000, qps: 28 },
  { time: '20:00', prompt_tokens: 210000, completion_tokens: 90000, qps: 16 }
]

const liveMockLatencyDistribution: LatencyBucket[] = [
  { label: '< 50ms (FAQ直出)', percent: 41.8, count: 14328, sub_label: '极速缓存命中' },
  { label: '50-200ms (小切片)', percent: 22.4, count: 7678, sub_label: '精确索引匹配' },
  { label: '200-500ms (多跳)', percent: 18.2, count: 6238, sub_label: '混合重排解析' },
  { label: '500ms-1s (重排)', percent: 12.1, count: 4147, sub_label: '复杂语义召回' },
  { label: '> 1s (长上下文)', percent: 5.5, count: 1889, sub_label: '大模型推理消耗' }
]

const liveMockTopRankings: TopRankingsResponse = {
  top_queries: [
    { rank: 1, query: '差旅与住宿报销标准及额度', count: 142 },
    { rank: 2, query: '跨境清关关税与港口延误预警', count: 118 },
    { rank: 3, query: '年假与带薪病假折算规则', count: 96 },
    { rank: 4, query: '研发内网 VPN 与堡垒机权限', count: 84 },
    { rank: 5, query: '供应商准入资质与付款账期', count: 72 }
  ],
  top_knowledge: [
    { rank: 1, title: '《员工差旅与住宿报销管理规范》', count: 186 },
    { rank: 2, title: '《跨境电商出口清关与税务政策指南》', count: 145 },
    { rank: 3, title: '《企业考勤管理与法定假期实施办法》', count: 129 },
    { rank: 4, title: '《研发基础架构与内网安全访问指引》', count: 112 },
    { rank: 5, title: '《供应链采购流程与供应商管理办法》', count: 98 }
  ]
}

const liveMockAuditLogs: AuditLogItem[] = [
  {
    id: 1,
    trace_id: 'trace-kg-10086-q01',
    created_at: '2026-09-24 14:23:05',
    time: '14:23:05',
    user_id: 1001,
    username: '张三',
    dept_name: '市场营销部',
    role_name: '普通员工',
    client_endpoint: 'Web 问答工作台',
    query: '请告诉我副总裁级别的期权激励折算系数是多少？',
    recalled_count: 3,
    allowed_count: 2,
    restricted_count: 1,
    is_blocked: true,
    verdict_type: 'BLOCKED',
    verdict_badge_text: '阻断 1 (4D越权) · 放行 2',
    tokens: 1120,
    latency_ms: 380,
    silent_intercept_triggered: true,
    sha256_hash: '8f9a2b77c3e5d0124a9f9394bf34a8e29d749a2ef778bca64e89139589d4d1e',
    chunk_verdicts: [
      {
        chunk_id: 'Chunk-101',
        chunk_name: 'Chunk-101: 《财务审批权限管理办法》 4.2节',
        doc_code: 'KU-1001',
        doc_title: '财务审批权限管理办法.docx',
        status: 'ALLOWED',
        checks: {
          global_public: true,
          dept_matched: true,
          role_matched: true,
          user_matched: true
        }
      },
      {
        chunk_id: 'Chunk-102',
        chunk_name: 'Chunk-102: 《2024年度薪酬分配基本总则》',
        doc_code: 'KU-1003',
        doc_title: '2024年度薪酬分配基本总则.pdf',
        status: 'ALLOWED',
        checks: {
          global_public: true,
          dept_matched: true,
          role_matched: true,
          user_matched: true
        }
      },
      {
        chunk_id: 'Chunk-103',
        chunk_name: 'Chunk-103: 高管期权折算系数与特别激励授予条件',
        doc_code: 'KU-1002',
        doc_title: '集团核心高管中长期薪酬与股权激励细则.docx',
        status: 'BLOCKED',
        checks: {
          global_public: false,
          dept_matched: false,
          dept_actual: '市场营销部',
          dept_target: '人力资源中心',
          role_matched: false,
          role_actual: '普通员工',
          role_target: '核心高管/HRBP',
          user_matched: false
        },
        reason: 'UNAUTHORIZED_DEPARTMENT_AND_ROLE: 所属部门与角色均未在授权策略内'
      }
    ],
    waterfall: [
      { step_name: '网关鉴权与会话校验', latency_ms: 12 },
      { step_name: '向量混合检索召回', latency_ms: 85 },
      { step_name: '4D-RBAC 权限判定硬过滤', latency_ms: 18 },
      { step_name: 'LLM 安全脱敏流式生成', latency_ms: 265 }
    ]
  },
  {
    id: 2,
    trace_id: 'trace-kg-10085-q04',
    created_at: '2026-09-24 14:18:22',
    time: '14:18:22',
    user_id: 1002,
    username: '王浩',
    dept_name: '基础架构部',
    role_name: '研发工程师',
    client_endpoint: 'Web 问答工作台',
    query: 'Apollo 生产集群故障转移应急操作手册与主备切换规程',
    recalled_count: 4,
    allowed_count: 4,
    restricted_count: 0,
    is_blocked: false,
    verdict_type: 'ALLOWED',
    verdict_badge_text: '✓ 4D全部放行',
    tokens: 2450,
    latency_ms: 310,
    silent_intercept_triggered: false,
    sha256_hash: '3e41b9c79234857ef09a47120a4bcfd39589a10385920dae783459cba671b2ef',
    chunk_verdicts: [
      {
        chunk_id: 'Chunk-201',
        chunk_name: 'Chunk-201: Apollo 集群高可用部署拓扑架构',
        doc_code: 'KU-2001',
        doc_title: 'Apollo运维操作指引.md',
        status: 'ALLOWED',
        checks: {
          global_public: false,
          dept_matched: true,
          role_matched: true,
          user_matched: true
        }
      },
      {
        chunk_id: 'Chunk-202',
        chunk_name: 'Chunk-202: 主备机房故障检测与秒级自动切换流程',
        doc_code: 'KU-2001',
        doc_title: 'Apollo运维操作指引.md',
        status: 'ALLOWED',
        checks: {
          global_public: false,
          dept_matched: true,
          role_matched: true,
          user_matched: true
        }
      }
    ],
    waterfall: [
      { step_name: '网关鉴权与会话校验', latency_ms: 10 },
      { step_name: '向量混合检索召回', latency_ms: 68 },
      { step_name: '4D-RBAC 权限判定硬过滤', latency_ms: 14 },
      { step_name: 'LLM 安全脱敏流式生成', latency_ms: 218 }
    ]
  },
  {
    id: 3,
    trace_id: 'trace-kg-10084-q02',
    created_at: '2026-09-24 14:12:08',
    time: '14:12:08',
    user_id: 1003,
    username: '张雪',
    dept_name: '市场营销部',
    role_name: '设计专员',
    client_endpoint: 'Web 问答工作台',
    query: '2024 年度新版品牌 VI 配色标准及字体下载授权路径',
    recalled_count: 1,
    allowed_count: 1,
    restricted_count: 0,
    is_blocked: false,
    verdict_type: 'FAQ_HIT',
    verdict_badge_text: '⚡ FAQ 命中直出',
    tokens: 0,
    latency_ms: 32,
    silent_intercept_triggered: false,
    sha256_hash: '12a9bc45ef7834091a457cba385901239845efacbd783920148593aef475892c',
    waterfall: [
      { step_name: '网关鉴权与会话校验', latency_ms: 8 },
      { step_name: 'Redis 极速缓存匹配', latency_ms: 14 },
      { step_name: '直出格式化封装', latency_ms: 10 }
    ]
  },
  {
    id: 4,
    trace_id: 'trace-kg-10083-q09',
    created_at: '2026-09-24 14:05:40',
    time: '14:05:40',
    user_id: 1004,
    username: '陈明',
    dept_name: '跨境电商部',
    role_name: '物流专员',
    client_endpoint: '移动端服务助手',
    query: '东南亚跨境电商清关延误预警与港口集装箱调度标准',
    recalled_count: 5,
    allowed_count: 5,
    restricted_count: 0,
    is_blocked: false,
    verdict_type: 'ALLOWED',
    verdict_badge_text: '✓ 4D全部放行',
    tokens: 3120,
    latency_ms: 480,
    silent_intercept_triggered: false,
    sha256_hash: '908475aebc4512987fec45672398450123948571029384756102938475610293',
    waterfall: [
      { step_name: '网关鉴权与会话校验', latency_ms: 15 },
      { step_name: '向量混合检索召回', latency_ms: 95 },
      { step_name: '4D-RBAC 权限判定硬过滤', latency_ms: 20 },
      { step_name: 'LLM 安全脱敏流式生成', latency_ms: 350 }
    ]
  },
  {
    id: 5,
    trace_id: 'trace-kg-10082-q03',
    created_at: '2026-09-24 13:58:19',
    time: '13:58:19',
    user_id: 1005,
    username: '赵峰',
    dept_name: '人力资源中心',
    role_name: '招聘专员',
    client_endpoint: 'Web 问答工作台',
    query: '集团离职补偿金核算细则与核心期权回购标准方案',
    recalled_count: 3,
    allowed_count: 1,
    restricted_count: 2,
    is_blocked: true,
    verdict_type: 'BLOCKED',
    verdict_badge_text: '阻断 2 (部门/角色越权)',
    tokens: 890,
    latency_ms: 390,
    silent_intercept_triggered: true,
    sha256_hash: '67483920148593aef475892c12a9bc45ef7834091a457cba385901239845efac',
    chunk_verdicts: [
      {
        chunk_id: 'Chunk-301',
        chunk_name: 'Chunk-301: 法定解除劳动合同经济补偿金基数规定',
        doc_code: 'KU-3001',
        doc_title: '员工离职管理制度.pdf',
        status: 'ALLOWED',
        checks: {
          global_public: true,
          dept_matched: true,
          role_matched: true,
          user_matched: true
        }
      },
      {
        chunk_id: 'Chunk-302',
        chunk_name: 'Chunk-302: 期权未归属权益清算与回购定价特别条款',
        doc_code: 'KU-1002',
        doc_title: '集团核心高管中长期薪酬与股权激励细则.docx',
        status: 'BLOCKED',
        checks: {
          global_public: false,
          dept_matched: true,
          dept_actual: '人力资源中心',
          dept_target: '人力资源中心',
          role_matched: false,
          role_actual: '招聘专员',
          role_target: '薪酬绩效专家/HRD',
          user_matched: false
        },
        reason: 'UNAUTHORIZED_ROLE: 当前账号角色非薪酬绩效专家或HR总监'
      }
    ],
    waterfall: [
      { step_name: '网关鉴权与会话校验', latency_ms: 11 },
      { step_name: '向量混合检索召回', latency_ms: 78 },
      { step_name: '4D-RBAC 权限判定硬过滤', latency_ms: 21 },
      { step_name: 'LLM 安全脱敏流式生成', latency_ms: 280 }
    ]
  }
]

// 1. 获取核心 KPI 卡片汇总
export const getDashboardSummaryApi = async (): Promise<ApiResponse<DashboardSummary>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 80))
    return {
      code: 200,
      message: 'success',
      data: liveMockSummary,
      trace_id: generateTraceId()
    }
  }

  return request.get('/api/v1/analytics/dashboard-summary')
}

// 2. 获取 Token 消耗与 QPS 走势
export const getTokenTrendsApi = async (
  days: number = 7
): Promise<ApiResponse<TokenTrendPoint[]>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 100))
    return {
      code: 200,
      message: 'success',
      data: liveMockTokenTrends,
      trace_id: generateTraceId()
    }
  }

  return request.get('/api/v1/analytics/trends', { params: { days } })
}

// 3. 获取响应延时区间分布
export const getLatencyDistributionApi = async (): Promise<ApiResponse<LatencyBucket[]>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 80))
    return {
      code: 200,
      message: 'success',
      data: liveMockLatencyDistribution,
      trace_id: generateTraceId()
    }
  }

  return request.get('/api/v1/analytics/latency-distribution')
}

// 4. 获取 TOP 5 双榜单 (高频提问与热门知识)
export const getTopRankingsApi = async (): Promise<ApiResponse<TopRankingsResponse>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 90))
    return {
      code: 200,
      message: 'success',
      data: liveMockTopRankings,
      trace_id: generateTraceId()
    }
  }

  return request.get('/api/v1/analytics/top-rankings')
}

// 5. 分页查询安全审计流水日志
export const getAuditLogsApi = async (
  params: PaginationParams & { status?: string; search?: string; keyword?: string }
): Promise<ApiResponse<PaginatedData<AuditLogItem>>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 120))
    let filtered = [...liveMockAuditLogs]
    if (params.status && params.status !== 'ALL') {
      if (params.status === 'BLOCKED') {
        filtered = filtered.filter((i) => i.is_blocked)
      } else if (params.status === 'ALLOWED') {
        filtered = filtered.filter((i) => !i.is_blocked && i.verdict_type !== 'FAQ_HIT')
      } else if (params.status === 'FAQ_HIT') {
        filtered = filtered.filter((i) => i.verdict_type === 'FAQ_HIT')
      }
    }
    const kw = params.search || params.keyword
    if (kw && kw.trim()) {
      const q = kw.trim().toLowerCase()
      filtered = filtered.filter(
        (i) =>
          i.trace_id.toLowerCase().includes(q) ||
          i.query.toLowerCase().includes(q) ||
          i.username.toLowerCase().includes(q) ||
          i.dept_name.toLowerCase().includes(q)
      )
    }

    const page = params.page || 1
    const pageSize = params.page_size || 10
    const start = (page - 1) * pageSize
    const paginatedItems = filtered.slice(start, start + pageSize)
    return {
      code: 200,
      message: 'success',
      data: {
        items: paginatedItems,
        total: filtered.length,
        page,
        page_size: pageSize
      },
      trace_id: generateTraceId()
    }
  }

  const queryParams: Record<string, unknown> = {
    page: params.page || 1,
    page_size: params.page_size || 10
  }
  if (params.status && params.status !== 'ALL') queryParams.status = params.status
  const kw = params.search || params.keyword
  if (kw && kw.trim()) queryParams.keyword = kw.trim()

  return request.get('/api/v1/analytics/audit-logs', { params: queryParams })
}

// 6. 获取审计证据链详情
export const getAuditLogDetailApi = async (traceId: string): Promise<ApiResponse<AuditLogItem>> => {
  if (isMockEnabled()) {
    await new Promise((resolve) => setTimeout(resolve, 80))
    const item = liveMockAuditLogs.find((i) => i.trace_id === traceId) || liveMockAuditLogs[0]
    return {
      code: 200,
      message: 'success',
      data: item,
      trace_id: generateTraceId()
    }
  }

  return request.get(`/api/v1/analytics/audit-logs/${traceId}`)
}
